### PR TITLE
Use well-defined signature scheme enum instead of arbitrary byte flag

### PR DESCRIPTION
**Pull request by patrickkuo**
This PR replaces the signature flag byte with enum.

Somehow TS SDK checks is skipped for this PR... @666lcz @punwai , are there anything needed to be changed in the TS SDK?

### CODE REVIEW

**Comment by patrickkuo** [comment]
friendly nudge.... this PR need some review 😉

**Comment by 666lcz** [comment]
@patrickkuo , yes the TS SDK needs to be updated. Will push some changes today

**Comment by punwai** [comment]
Sorry for this, I've been on leave to attend a conference-ish event. I've had a look at this and looks good to me on the RPC and crypto.rs side. Although Chris would probably know more about the TS side.

File: crates/sui-gateway/src/unit_tests/rpc_server_tests.rs [hunk 1]
```diff
@@ -150,12 +150,12 @@ async fn test_move_call() -> Result<(), anyhow::Error> {
     let signature = keystore.sign(address, &tx_bytes)?;
 
     let tx_bytes = Base64::from_bytes(&tx_bytes);
-    let flag_bytes = Base64::from_bytes(&[signature.flag_byte()]);
+    let sig_scheme = signature.scheme();
```
Comment (punwai) [inline crates/sui-gateway/src/unit_tests/rpc_server_tests.rs:153]: Thanks for this, it is much clearer this way!

File: crates/sui-open-rpc/spec/openrpc.json [hunk 2]
```diff
@@ -160,11 +160,11 @@
           }
         },
         {
-          "name": "flag",
+          "name": "sig_scheme",
           "description": "Flag of the signature scheme that is used.",
```
Comment (punwai) [inline crates/sui-open-rpc/spec/openrpc.json:164]: Would it be possible to specify that eddsa is 0x00? (If it isn't already)

**Review by punwai — APPROVED** [review]

File: crates/sui-open-rpc/spec/openrpc.json (hunk 2 above)
Comment (patrickkuo) [reply crates/sui-open-rpc/spec/openrpc.json:164]: I can add that in the description

**Comment by patrickkuo** [comment]
Thanks! @punwai  and @666lcz , I will leave this open for the TS SDK changes

**Review by kchalkias — APPROVED** [review]
Thanks Patrick!

**Comment by 666lcz** [comment]
Updated the TS side and verified that transaction execution works on a local network

**Review by 666lcz — APPROVED** [review]

**Comment by patrickkuo** [comment]
> Updated the TS side and verified that transaction execution works on a local network

Thanks @666lcz !!

**Pull request merged**